src/methods/BaseCompare.js

```diff
@@ -126,12 +126,13 @@ export default class BaseCompare {
     return path.join(runtimeConfigPath, `${name}-${suffix}.json`);
   }
 
-  createResultReport(misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions) {
+  createResultReport(misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions, options = {}) {
     return {
       misMatchPercentage,
       isWithinMisMatchTolerance,
       isSameDimensions,
-      isExactSameImage: misMatchPercentage === 0
+      isExactSameImage: misMatchPercentage === 0,
+      ...options
```

Comment:
Nit: to be safe, wouldn't we want to spread `options` first? In the future, we could potentially override the other object properties if we add duplicate properties to the passed in `fileNames` object.

Reply:
```javascript
return {
  ...options, //spread this first
  misMatchPercentage,
  isWithinMisMatchTolerance,
  isSameDimensions,
  isExactSameImage: misMatchPercentage === 0,
};
```
Just to be safe. If `options` somehow contains any other keys (`misMatchPercentage`, `isWithinMismatchTolerance`, `isSameDimensions`, `isExactSameImage`), then spreading `options` last would override those values.

Reply:
Gotcha.